gradlew

```diff
@@ -163,10 +163,18 @@ fi
 
 CLASSPATH=$GRADLE_WRAPPER_JAR
 
-# Don't fork a daemon mode on initial run that generates local defaults.
-GRADLE_DAEMON_CTRL=
+# Generate gradle.properties if they don't exist
 if [ ! -e "$APP_HOME/gradle.properties" ]; then
-    GRADLE_DAEMON_CTRL=--no-daemon
+    echo "Populating gradle.properties from gradle.properties.template."
+    # Do the copy first. If the next steps fail, the gradle.properties will still be usable
+    cp gradle.properties.template gradle.properties
+    NUM_CORES=$(grep ^cpu\\scores /proc/cpuinfo | uniq |  awk '{print $4}')
+    if [ $NUM_CORES -ge 12 ]; then
+      NUM_CORES=12
+    fi
+    sed -i "s/org.gradle.workers.max=[[:digit:]]\+/org.gradle.workers.max=${NUM_CORES}/g;s/tests.jvms=[[:digit:]]\+/tests.jvms=${NUM_CORES}/g" gradle.properties
```

Comment:
This modification could be wrapped in a test for whether $NUM_CORES is in fact a positive integer, to avoid replacing default with invalid value.